src/contentmanagermodel.cpp

```diff
@@ -111,13 +71,13 @@ QModelIndex ContentManagerModel::parent(const QModelIndex &index) const
     if (!index.isValid())
         return QModelIndex();
 
-    auto childItem = getSharedPointer(static_cast<Node*>(index.internalPointer()));
-    if (!childItem)
-        return QModelIndex();
+    auto childItem = static_cast<Node*>(index.internalPointer());
     auto parentItem = childItem->parentItem();
 
     if (parentItem == rootNode)
         return QModelIndex();
+    if (!parentItem)
+        return QModelIndex();
```

Comment:
Better with :

```cpp
if (!parentItem || parentItem == rootNode) {
    return QModelIndex();
}
```

Reply:
Please fix/improve last detail and I will merge.